config/config.go

```diff
@@ -32,3 +42,37 @@ func AddFlags(cmd *cobra.Command) {
 		panic(err)
 	}
 }
+
+func PersistentStartupProcessFlags() error {
+
+	if err := convertResolveHostsToMap(); err != nil {
+		return err
+	}
+	return nil
+
+}
+
+func convertResolveHostsToMap() error {
+	for _, resolveHost := range viper.GetStringSlice(optname.Resolve) {
+		split := strings.SplitN(resolveHost, ":", 3)
+		if len(split) != 3 {
+			return fmt.Errorf("invalid resolve host format, expected <hostname>:port:<ip>, got: %s", resolveHost)
+		}
+		host, port, addr := strings.ToLower(split[0]), split[1], split[2]
+		hostPort := fmt.Sprintf("%s:%s", host, port)
```

Comment:
```suggestion
		hostPort := net.JoinHostPort(host, port)
```
https://pkg.go.dev/net#JoinHostPort